src/main/java/TaskList.java

```diff
@@ -0,0 +1,124 @@
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class TaskList {
+
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    public enum TaskType {
+        DEADLINE,
+        EVENT,
+        TODO
+    }
+
+    public void addToList(TaskType taskType, String description, String deadline)
+            throws IllegalArgumentException {
+        Task newTask = null; // task to be added
+
+        // based on task type, create corresponding subclass object
+        switch (taskType) {
+        case DEADLINE:
+            newTask = new Deadline(description, deadline);
+            break;
+        case EVENT:
+            newTask = new Event(description, deadline);
+            break;
+        case TODO:
+            newTask = new Todo(description);
+            break;
+        default:
+            // this should not happen; invalid task type should be caught in caller method already
+            System.out.println("Something went wrong!");
+            break;
+        }
+
+        // add newTask to taskList
+        tasks.add(newTask);
+        System.out.println("Okay! I've added: " + description
+                + ". Use list to see all your tasks!");
+        // todo: description has trailing space due to using /at or /by as delimiter. how?
+    }
+
+    public void printList() {
+        int i = 1;
+        for (Iterator<Task> iterator = this.tasks.iterator(); iterator.hasNext(); i++) {
+            String s = iterator.next().toString();
+            System.out.println(i + ". " + s);
+        }
+    }
+
+    public Task getTask(int taskIndex) throws IndexOutOfBoundsException {
+        return tasks.get(taskIndex - 1);
+    }
+
+    /**
+     * Processes the given input string.
+     *
+     * @param input input string given by user.
+     */
+    public void processInput(String input) throws InputMismatchException {
```

Comment:
Instead of having almost all the app logic here, maybe it would help to define a `Command` class and a few subclasses like `AddCommand`, `DoneCommand`, etc. It would make the code more maintainable.